loader/loader.go

```diff
@@ -379,7 +385,7 @@ func loadYamlModel(ctx context.Context, config types.ConfigDetails, opts *Option
 			}
 
 			if opts.Interpolate != nil && !opts.SkipInterpolation {
-				cfg, err = interp.Interpolate(cfg, *opts.Interpolate)
+				cfg, err = interp.Interpolate(cfg, interpolateOptsBeforeMerge)
```

Comment:
I wonder:
as extends will always require interpolation, wouldn't it be simpler to just force interpolation in `ApplyExtends` and `ApplyInclude`?

Reply:
1. We call `ApplyExtends` only if `SkipExtends` is not `true` while interpolation happens always(the same for Include)
2. We call `ApplyInclude` after interpolation and `processors`, if we start interpolate inside `ApplyInclude` in can have consequences, I wanted to keep current logic as much as I can